pandas/plotting/_core.py

```diff
@@ -843,11 +844,16 @@ def _plot_colorbar(self, ax, **kwds):
         # For a more detailed description of the issue
         # see the following link:
         # https://github.com/ipython/ipython/issues/11215
-
         img = ax.collections[0]
         cbar = self.fig.colorbar(img, ax=ax, **kwds)
+
+        if _mpl_ge_3_0_0():
+            # The workaround below is no longer necessary.
```

Comment:
could xfail?

Reply:
This is in the plotting code itself, not in the test. The workaround is necessary for mpl < 3.0, but breaks things in mpl >= 3.0